src/engine/index.ts

```diff
@@ -20,4 +24,61 @@ const check = (
     : !checkedConditions.includes(false);
 };
 
-export { translate, check };
+const doesAudienceVersionMatch = (
+  audience: AudienceDefinition,
+  matchedAudience: MatchedAudience
+) =>
+  audience.id === matchedAudience.id &&
+  audience.version === matchedAudience.version;
+
+const engine = (
+  audienceDefinitions: AudienceDefinition[],
+  matchedAudienceStore: MatchedAudienceStore,
+  viewStore: ViewStore
```

Comment:
The engine does not need to depend on the implementations of the store, I would prefer:

```javascript
const audiences = getAudiences()
const views = store.getViews()
const matched = engine.check(audiences, views)
store.setMatched(matched)
```

Over

```javascript
engine(ViewsStore, AudienceStore)
```
This means engine needs to know all about the implementation of the two stores, it should only need to receive data in a specific format and do its work. The second version also reads much worse from a flow perspective.

Reply:
I agree, but [there is coupling](https://github.com/AirGrid/edgekit/pull/130/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L38-L42) preventing the suggested implementation from working.
I'll try to rewrite it a little more aggressively to see if I can remove it, then.

Reply:
There is no need to re-write, it was already like this.

I am thinking I am just going to revert to an earlier version without all the refactor you introduced. Sorry but this has again become a very long process for what is quite a small feature.